grpc/common/common.go

```diff
@@ -155,6 +155,18 @@ func ToCoreMetric(mt *Metric) core.Metric {
 	return ret
 }
 
+func MetricToRequested(mts []*Metric) []core.RequestedMetric {
+	ret := []core.RequestedMetric{}
```

Comment:
This slice should have length of `len(mts)` to avoid unnecessary reallocs. Similar issue in following function.